### DISCUSSION TITLE
Toggle multiple things with one midi button?

### DISCUSSION THREAD

**Discussion by BeardDesign1** [cpyarger/obs-midi#104] [Ideas]
**Is your feature request related to a problem? Please describe.**
I'm trying to disable one filter and enable another by tapping the same pad, right now I have a workaround using note on for one and note off for the other but it's not ideal, also would like to both switch to a wait scene and mute my mic with one button but again that's not really possible.

**Describe the solution you'd like**
Letting people add multiple actions for the same midi button with the same message type.

**Describe alternatives you've considered**
Maybe having a "Multiple actions" action that opens up a window with gui to add multiple things? that might avoid clogging up the configure tab.

Is that what the macro page is eventually going to be for? If it is and I'm just dumb then ignore this lol

**Comment by cpyarger** [comment]
That is what the macros page will be for (:

**Comment by cpyarger** [comment]
basically the macros page will allow you to build a macro, or multiple action sequence. including adding delays and repeat actions. Right now the macros page is there as a teaser of the upcoming feature. though you shouldnt be able to do anything with it. 

This plus the device feedback mapping system are currently in development on the side. and will be released for test once they are in a working state. 

the idea for the workflow of the macros, is set a macro name, then add multiple actions into the macro, then in the config tab you will see in the actions dropdown a selector for the macros, and another dropdown will appear giving you the current list of macros to choose from.

**Comment by cpyarger** [comment]
Moved this to here as this is where discussions go. I would love any other feedback or ideas you have

**Comment by BeardDesign1** [comment]
ohh okay then, looking forward to macros

**Comment by BeardDesign1** [comment]
As for suggestions, being able to sort by action type, scene, etc by clicking on the columns would be very useful but that's not really a pressing issue

**Reply by cpyarger** [reply]
in the mapping table?

**Reply by BeardDesign1** [reply]
yes to get a clearer view of the whole thing, if I keep making new scenes and adding shortcuts for those new scenes I can see myself getting lost in there haha

**Reply by cpyarger** [reply]
I'll see what I can do.
